src/frontend/flask_server.py

```diff
@@ -146,21 +146,26 @@ def payment():
             label = request.form.get('contact_label', None)
             if label:
                 # new contact. Add to contacts list
-                _add_contact(label,
-                             recipient,
-                             LOCAL_ROUTING,
-                             False)
+                response = _add_contact(label,
+                                        recipient,
+                                        LOCAL_ROUTING,
+                                        False)
+                if response.status_code == 409:
+                    # contact not allowed.
+                    msg = 'Payment failed: ' + response.json()['msg']
+                    return redirect(url_for('home', msg=msg))
```

Comment:
- It might be better to throw an exception in _add_contact and catch it with the others
- Should we only stop the transaction if _add_contact returns 409? What if the contact service throws a different error? I'd say we should treat those cases the same

Reply:
Good idea!  Both points done.